### ISSUE TITLE
What does frequency mean ?

### ISSUE THREAD

**Issue by arunnbabu81** [trumad/budgetlywiki#1]
First of all thanks for creating budgetlywiki. I was trying to use it in my wiki for tracking my income and expenses.

I saw that `all calculations are done on a per-month basis`. What does this means. Is it that, after the current month (April), all the calculations of April will change and will start once again from the beginning for the month of May?

**Comment by trumad** [comment]
Thank you for using it! I'm sorry, I meant to write better documentation for this, and maybe even make a video.

It's actually not even as smart as that. The "frequency" tiddler can be set to any word you like. "month" is the default, but you could change it to say "week" and then you just need to make sure that every income/expense is the amount that you receive/pay in a week. So there's no logic happening really. It's just there to remind you that if you're inputting an expense that you've calculated it to be a monthly expense.

For example, if you pay Disney Plus annually, you should divide that amount by 12 and then input that result into a Disney Plus tiddler. Then I'd probably tag it with the account "occasional spending", and make sure you transfer into that account $Result per month, so that that account has the full amount in it whenever the DIsney Plus payment is due.

I hope that makes sense... Basically you just use this calculator so you know how much money to move into which accounts when your income arrives, and then when the money is going out as expenses you will be confident that each account has enough money in it to cover your expenses. The assumption is that you get paid monthly, and you just move your money around on the 1st of the month or whatever, but if you get paid at a weekly "frequency" then you'll need to do some math manually to make the numbers line up. If your phone bill comes out once a month, but you get paid weekly, multiply your phone bill by 12 and divide by 52, and put that result in your phone bill tiddler, so each week you're putting enough money into the right place to cover that bill when it arrives.

**Comment by arunnbabu81** [comment]
Is there any way to accumulate the delta (or income or expense) calculated for each month, so that we get a idea of the yearly income and spending.

**Comment by trumad** [comment]
if you change the text of the budgetDelta tiddler to this, you'll get the yearly delta instead of the monthly delta:

`(= ({{Income Total}} - {{Expenses Total}})*12 =)`

Then try unzipping the attached file and drag the .tid file into your tiddlywiki. Then edit the Budget Overview tiddler so that instead of `<<listOfList Budget>>`, it's got `<<listOfListAnnual Budget>>`. Again, this is hard-coded to assume that you're entering monthly expenses for all the other tiddlers.

[listAnnualBudget.tid.zip](https://github.com/trumad/budgetlywiki/files/8496406/listAnnualBudget.tid.zip)

**Comment by arunnbabu81** [comment]
Thanks for the help. 

What will you do for spending like Fuel, stationaries which won't be the same for each month.

**Comment by trumad** [comment]
Yeah, that's kinda guesswork. Maybe look at your fuel expenses over the past 6 months, multiply by 2 and divide by 12. Then as long as you put that amount in per month, you'll know roughly what to expect. Of course these days you'll need to account for the rise in fuel expenses when guessing the monthly amount 🙄 

Nice to know that somebody is actually using this thing. I'll try to take some time to write some better documentation for it, or improve the logic a bit.

**Comment by arunnbabu81** [comment]
> Nice to know that somebody is actually using this thing.

I came to know about this from the talk tiddlywiki forum

https://talk.tiddlywiki.org/t/budgetlywiki-tiddlywiki-for-managing-your-budget/1123


> I'll try to take some time to write some better documentation for it, or improve the logic a bit.

This is the only other TW I know which deal with expenses. 
https://ooktech.com/TiddlyWiki/ExpenseReports/    But it is not actively maintained.
This might be of help.
